src/meta.jl

```diff
@@ -266,36 +265,93 @@ end
-    func_name = pm_name(pf)
-    app = pm_app(pf)
-
+function jl_function(callable::Symbol,
+                     jl_name::Symbol,
+                     pm_name::String,
+                     app_name::String)
     return quote
-        function $(jlfunc_name)(::Type{PropertyValue}, args...;
+        function $(jl_name)(::Type{PropertyValue}, args...;
             template_parameters::Base.AbstractVector{<:AbstractString}=String[],
             kwargs...)
-
-            return $(callable(pf))(PropertyValue, Symbol($app), Symbol($func_name), args...;
+            return $(callable)(PropertyValue, Symbol($app_name), Symbol($pm_name), args...;
                 template_parameters=template_parameters, kwargs...)
         end;
-        function $(jlfunc_name)(args...;
+        function $(jl_name)(args...;
             template_parameters::Base.AbstractVector{<:AbstractString}=String[],
             kwargs...)
-
-            return $(callable(pf))(Symbol($app), Symbol($func_name), args...;
+            return $(callable)(Symbol($app_name), Symbol($pm_name), args...;
                 template_parameters=template_parameters, kwargs...)
         end;
-        function $(Base.Docs).getdoc(::typeof($(jlfunc_name)))
-            return PolymakeDocstring($(doc_string))
-        end;
+    end
+end
+
+function jl_function(callable::Symbol,
+                     jl_name::Symbol,
+                     pm_name::String,
+                     app_name::String,
+                     templates)
+    return quote
+        function $(jl_name){$(templates...)}(::Type{PropertyValue},
+            args...;
+            kwargs...) where {$(templates...)}
+            template_parameters = translate_type_to_pm_string.([$(templates...)])
+            return $(callable)(PropertyValue, Symbol($app_name), Symbol($pm_name), args...;
+                template_parameters=template_parameters, kwargs...)
+        end
+        function $(jl_name){$(templates...)}(args...; kwargs...) where {$(templates...)}
+            template_parameters = translate_type_to_pm_string.([$(templates...)])
+            return $(callable)(Symbol($app_name), Symbol($pm_name), args...;
+                template_parameters=template_parameters, kwargs...)
+        end
+    end
+end
+
+jl_function(pf::PolymakeFunction, templates) =
+    jl_function(callable(pf), jl_symbol(pf), pm_name(pf), pm_app(pf), templates)
+
+jl_function(pf::PolymakeFunction) =
+    jl_function(callable(pf), jl_symbol(pf), pm_name(pf), pm_app(pf))
+
+function jl_code(pf::PolymakeFunction, doc_string=docstring(pf))
+    jlfunc_name = jl_symbol(pf)
+    func_name = pm_name(pf)
+    app = pm_app(pf)
+    min_tparam = mandatorytemplates(pf)
+
+    if istemplated(pf)
+        tparams = Symbol.(map(x->get(x,"name",String),templates(pf)))
+        templated_functions= [
+            jl_function(pf, tparams[1:i])
+               for i in max(min_tparam,1):length(tparams)
+        ]
+
+        functiondef = quote
```

Comment:
i see, we create a faux type for this functionality

Reply:
does it hurt too much?

We already won the highest startup time contest for Oscar.

Reply:
either way, I'm fine with this :)

Reply:
I can see one (tiny) drawback: now `polytope.cube` will explain itself as type and you need to dig deeper to see that it's a function:
```julia
julia> polytope.cube
Polymake.polytope.cube


julia> methods(polytope.cube)
# 2 methods for type constructor:
[1] (::Type{Polymake.polytope.cube})(::Type{Polymake.PropertyValue}, args...; template_parameters, kwargs...) in Polymake.polytope at /home/kalmar/.julia/dev/Polymake/src/meta.jl:276
[2] (::Type{Polymake.polytope.cube})(args...; template_parameters, kwargs...) in Polymake.polytope at /home/kalmar/.julia/dev/Polymake/src/meta.jl:282

```

---
that's because we have two versions of jit to initialize:P